viewer/packages/360-images/src/icons/IconCollection.ts

```diff
@@ -17,12 +22,44 @@ export class IconCollection {
   private readonly _icons: Image360Icon[];
   private readonly _iconsSprite: InstancedIconSprite;
   private readonly _computeClustersEventHandler: BeforeSceneRenderedDelegate;
+  private readonly _computeProximityPointsEventHandler: BeforeSceneRenderedDelegate;
   private readonly _onBeforeSceneRenderedEvent: EventTrigger<BeforeSceneRenderedDelegate>;
 
+  private _activeCullingSchemeEventHandeler: BeforeSceneRenderedDelegate;
+  private _iconCullingScheme: IconCullingScheme;
+  private _proximityRadius = 0;
+  private _proximityPointLimit = 50;
+
   get icons(): Image360Icon[] {
     return this._icons;
   }
 
+  public setCullingScheme(scheme: IconCullingScheme): void {
+    if (this._iconCullingScheme === scheme) return;
+
+    this._iconCullingScheme = scheme;
+    this._onBeforeSceneRenderedEvent.unsubscribe(this._activeCullingSchemeEventHandeler);
+
+    switch (this._iconCullingScheme) {
+      case IconCullingScheme.Clustered: {
+        this._activeCullingSchemeEventHandeler = this._computeClustersEventHandler;
+        break;
+      }
+      case IconCullingScheme.Proximity: {
+        this._activeCullingSchemeEventHandeler = this._computeProximityPointsEventHandler;
+        break;
+      }
+      default:
+        break;
+    }
+    this._onBeforeSceneRenderedEvent.subscribe(this._activeCullingSchemeEventHandeler);
+  }
+
+  public set360IconCullingRestrictions(radius: number, pointLimit: number): void {
+    this._proximityRadius = radius;
```

Comment:
Should probably do a min clamp here. I.E. `Math.min(0, radius)`, as negative values would probably create undefined behaviour / we would not need to handle / think about it.

Reply:
Done! But with max 💯

Reply:
I think you should remove the max cap, we can have point clouds that span kilometers, so in those cases if they only have a few 360 images, but they span great areas, I would still like to see them.